sn_cli/src/subcommands/register.rs

```diff
@@ -6,31 +6,44 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
+use bls::PublicKey;
 use clap::Subcommand;
 use color_eyre::Result;
 use sn_client::{Client, ClientRegister, Error as ClientError};
+use sn_protocol::storage::RegisterAddress;
 use xor_name::XorName;
 
 #[derive(Subcommand, Debug)]
 pub enum RegisterCmds {
-    /// Create a new register with the given pet name.
+    /// Create a new register with a name.
     Create {
-        /// The name of the register to create.
+        /// The name of the register to create. This could be the app's name.
+        /// This is used along with your public key to derive the address of the register
         #[clap(name = "name")]
         name: String,
     },
     Edit {
-        /// The name of the register to edit.
-        #[clap(name = "name")]
-        name: String,
+        /// The address of the register to edit.
+        #[clap(name = "address")]
+        address: String,
+        /// If you are the owner, the name of the register can be used as a shorthand to the address,
```

Comment:
the address is deduced from the `name + pk` , which, as long as the pk is known, anyone can deduce the address of a register from its name ?

Reply:
That is correct. That's why the owner of a register can use the name directly in the cli while the others will need to use the address because their own key is different. 
This also means anyone can get the owner's key from the address, although we could hide it with a hash and a derived key, but that would add complexity, let's keep it simple for now.